File: example/loop/loop.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

/* sum the numbers from 0 to n-1 */
func sum(n uintptr) uintptr {
	var ret uintptr
	for i := uintptr(0); i < n; i++ {
		ret += i
	}
	return ret
}

func main() {
	n := uintptr(1000000000)
	if len(os.Args) > 1 {
		n2, err := strconv.ParseUint(os.Args[1], 10, 64)
		if err == nil {
			n = uintptr(n2)
		}
	}
	fmt.Println(sum(n))
}

File: example/loop/_loop.c
#include <stddef.h>
#include <stdlib.h>
#include <stdio.h>

/* sum the numbers from 0 to n-1 */
size_t sum(size_t n) {
    size_t i, ret = 0;
    for (i = 0; i < n; i++) {
        ret += i;
    }
    return ret;
}

int main(int argc, char* argv[]) {
    size_t n = 1000000000;
    if (argc > 1) {
        n = strtoul(argv[1], NULL, 10);
    }
    printf("%lu\n", (unsigned long)sum(n));
}
